src/Compilers/Core/Portable/PEWriter/ReferenceIndexer.cs

```diff
@@ -119,7 +119,7 @@ protected override void ProcessMethodBody(IMethodDefinition method)
                         }
                     }
                 }
-                else if (!metadataWriter.allowMissingMethodBodies)
+                else if (!metadataWriter.metadataOnly)
```

Comment:
Good point. I don't think we should process an existing body or even get such a body when metadataOnly. In that case I don't think we should call `method.GetBody` at all.

---
In reply to: [105751241](https://github.com/dotnet/roslyn/pull/17558#discussion_r105751241) [](ancestors = 105751241)